src/__test__/redux/actions/experiments/switchExperiment.test.js

```diff
@@ -0,0 +1,21 @@
+import { switchExperiment } from 'redux/actions/experiments';
+import rootReducer from 'redux/reducers/index';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import cellSetsInitialState from 'redux/reducers/cellSets/initialState';
+import experimentsInitialState from 'redux/reducers/experiments/initialState';
+
+const storeState = require('__test__/data/mock_store.json');
+
+describe('switch experiment ', () => {
+  it('switches the experiment to its initial values', () => {
+    const store = createStore(rootReducer, storeState, applyMiddleware(thunk));
+    store.dispatch(switchExperiment());
+    expect(store.getState().backendStatus).toEqual(storeState.backendStatus);
+    expect(store.getState().experimentSettings).toEqual(storeState.experimentSettings);
+    expect(store.getState().samples).toEqual(storeState.samples);
+    expect(store.getState().projects).toEqual(storeState.projects);
+    expect(store.getState().cellSets).toEqual(cellSetsInitialState);
+    expect(store.getState().experiments).toEqual(experimentsInitialState);
+  });
```

Comment:
Hmm... This test doesn't look like it does anything. You haven't' made any modifications to the store that would allow us whether the store is really reset or not.

Reply:
Can we use the real store (via `makeStore` instead of bringing `createStore`?) that would greatly involve other codes in our platform too.

Reply:
Why should we load everything like we are  testing a component. This is a simple action to reset some things to their initial state which uses the actual file thats used and keep the value of the selected store parts as is? Initially i was going to do that  dispatch loading and everything, but it just seemed unnecessarily complex when this small functionality can be tested in a simple way. What do you think? Also I created a file which has all the values in the store so we can check by that.

Reply:
Ooh I see. I didn't understand that `storeState` actually contains a store that's loaded with experiments, projects, etc. Can you rename `storeState` to something more descriptive (e.g. `storeWithExperimentData`)?

And yea, I think instead of checking that the properties equal to their initial state one by one, you can use `matchSnapshot` because there are more properties which you have not checked for (e.g. embeddings, genes, cellMeta, cellInfo, etc.)